hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -49,12 +49,14 @@
    */
   public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
     List<HCatFieldSchema> fields = Lists.newArrayList();
+    //TODO: Finalize a consistent way to whether do projection on source or not while reading and if needed
+    // do type check here
     for (Schema.Field field : schema.getFields()) {
       String name = field.getName();
       try {
         // this field of the schema must exist in the table
         Preconditions.checkNotNull(tableSchema.get(name), "Missing field %s in table schema", name);
-        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema().getType()), ""));
+        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema()), ""));
```

Comment:
The inconsistent/unknown part is whether we should take schema(field and type) from the user or just the fields (like database plugin). After the above comments I am assuming we want to take the schema so I will add the type check part.